src/Server.cpp

```diff
@@ -0,0 +1,211 @@
+#include "../inc/Server.hpp"
+
+/* ====== HELPER FUNCTIONS ====== */
+
+/**
+ * @brief Set the socket to nonblocking mode
+ * 
+ * Use fcntl to manipulate fd/sockets. Retrieve the currently set flags with
+ * F_GETFL. Set the flag with with F_SETFL. Use bitwise OR to set the 
+ * O_NONBLOCK bit in "flags" while preserving all other bits.
+ * @param sock	Socket to be manipulated
+ */
+static void setNonblocking(int sock)
+{
+	int	flags = fcntl(sock, F_GETFL, 0);
+	if (flags == -1) {
+		throw std::runtime_error("fcntl(F_GETFL)");
+	}
+	if (fcntl(sock, F_SETFL, flags | O_NONBLOCK) == -1) {
+		throw std::runtime_error("fcntl(F_SETFL)");
+	}
+}
+/* ====== CONSTRUCTOR/DESTRUCTOR ====== */
+
+/**
+ * @brief Constructor for the Server class.
+ * 
+ * This constructor initializes a server instance by creating a server socket,
+ * setting it to non-blocking mode, binding it to a specific address and port,
+ * listening on the server socket, creating an epoll instance for event handling,
+ * and adding the server socket to the epoll instance.
+ * 
+ * @throws std::runtime_error if any of the socket creation, binding, listening,
+ *         epoll creation, or epoll control operations fail.
+ * 
+ */
+Server::Server()
+{
+    // Create server socket using TCP protocol SOCK_STREAM
+	_serverSock = socket(AF_INET, SOCK_STREAM, 0);
+	if (_serverSock < 0) {
+		throw std::runtime_error("socket");
+	}
+
+	// Set server socket to non-blocking
+	setNonblocking(_serverSock);
+
+	// Bind server socket
+    struct sockaddr_in	server_addr = {};
+	server_addr.sin_family = AF_INET; // Communicates with IPv4
+	server_addr.sin_addr.s_addr = INADDR_ANY; // Accept any arriving ip addresses
+	server_addr.sin_port = htons(PORT); // Listen on specified port
+	if (bind(_serverSock, (struct sockaddr*)&server_addr, sizeof(server_addr))
+		< 0) {
+  		close(_serverSock);
+		throw std::runtime_error("bind");
+	}
+
+	// Listen on server socket
+	if (listen(_serverSock, 10) < 0) {
+        close(_serverSock);
+		throw std::runtime_error("listen");
+	}
+
+	// Create epoll instance
+	_epfd = epoll_create1(0);
+	if (_epfd < 0) {
+        close(_serverSock);
+		throw std::runtime_error("epoll_create1");
+	}
+
+	// Add server socket to epoll instance
+	_ev.events = EPOLLIN | EPOLLET; // Edge-triggered mode
```

Comment:
Yes, that's what I think as well. Shame though as it would be so much more elegant. We will need to jump back and forth all the time to check if there is a pending action.